src/Simplify_Call.cpp

```diff
@@ -355,14 +360,12 @@ Expr Simplify::visit(const Call *op, ExprInfo *bounds) {
         }
     } else if (op->is_intrinsic(Call::saturating_cast)) {
         internal_assert(op->args.size() == 1);
-        ExprInfo a_bounds;
-        Expr a = mutate(op->args[0], &a_bounds);
-
-        // TODO(rootjalex): We could be intelligent about using a_bounds to remove saturating_casts;
```

Comment:
Are you removing this TODO because you think it's not worth doing?

Reply:
I think converting a saturating cast to a cast is likely to mess up instruction selection later

Reply:
Ah, for ISAs that only support saturating casts? Maybe we need #7805 - style instruction selection to rewrite casts to saturating casts on those platforms